### ISSUE TITLE
BUG: Two Different Albums are being displayed as the same album.

### ISSUE THREAD

**Issue by xekon** [ampache/ampache#2667]
### Describe the bug
Two albums with the same name from different Artists are being mixed together as the same album.

If this is the byproduct of some clever script that tries to appropriately combine files that it thinks are the same album, is there a setting to disable it?

My files are very carefully tagged, these are clearly not the same album. (No Doubt and Scorpions, LOL!)

### To reproduce

1. I noticed this while browsing the albums.

I double checked the Tag data:

Scorpions - Icon - track 01:
```
Complete name                            : /media/md0/audio/Rock-Classic/Scorpions/2010 - Icon (HITS)/01 - Rock you like a hurricane.mp3
Format                                   : MPEG Audio
File size                                : 9.67 MiB
Duration                                 : 4 min 13 s
Overall bit rate mode                    : Constant
Overall bit rate                         : 320 kb/s
Album                                    : Icon
Track name                               : Rock you like a hurricane
Track name/Position                      : 01
Performer                                : Scorpions
Genre                                    : Classic Rock
Original/Released date                   : 2010
Recorded date                            : 2010
Writing library                          : LAME3.92 
```

No doubt - Icon - Track 01:
```
Complete name                            : /media/md0/audio/Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/01 - Just A Girl.mp3
Format                                   : MPEG Audio
File size                                : 7.86 MiB
Duration                                 : 3 min 25 s
Overall bit rate mode                    : Constant
Overall bit rate                         : 320 kb/s
Album                                    : Icon
Track name                               : Just A Girl
Track name/Position                      : 01
Performer                                : No Doubt
Genre                                    : Alternative Rock
Original/Released date                   : 2010
Recorded date                            : 2010
Writing library                          : LAME3.98r
```

### Expected behavior
These two Albums should not be displayed as the same album.

### Screenshots
![2020-12-26_22-49](https://user-images.githubusercontent.com/106644/103165427-9d20b900-47cc-11eb-8eab-2d8a3e33ca02.png)

![2020-12-26_23-30](https://user-images.githubusercontent.com/106644/103165926-75ccea80-47d2-11eb-9638-7f3b73033900.png)

## Environment

* Ampache version: Latest Develop clone from github master
* Web server + version: Nginx
* Server operating system: Ubuntu 20.04
* Client operating system: Ubuntu 20.04 with Firefox

**Comment by kuzi-moto** [comment]
For albums of the same name from different artists, using MBID is preferred, see the [FAQ](https://github.com/ampache/ampache/wiki/FAQ#albums-with-the-same-title-are-being-displayed-as-a-single-album) for more info.

**Comment by xekon** [comment]
I actually really dislike Music Brainz, I love the idea of it but every single time I have tried to use it I spend significantly more time than if I just used my custom wrote script that tags them according to path and filename structure.

Still, thanks for the info @kuzi-moto maybe I can ONLY the MBID tag and only on albums that I find with this problem. (That was the other thing I disliked, sometimes there is just far too much tag data added.)

**Comment by kuzi-moto** [comment]
Yes you are correct that you could add only MBID tags to the affected albums.

There can be a lot of data, but I'm not sure that's a bad thing? Tag data compared to song data is a tiny fraction. The different tagging software should allow you to add only which tags you want. Not sure most people would run into this, but MBID also allows for keeping multiple releases of the same album.

**Comment by xekon** [comment]
hmmm so i am reading about the MBID, there are apparently many:

    the recording itself
    the release
    the label
    the track artist
    the release artist

which MBID do I need? going to see if I can find the section of code that talks about it.

**Comment by xekon** [comment]
The FAQ that you posted says:

"This is a known limitation; the alternative is for Various Artist albums to display as multiple albums, which is unacceptable."

I would like to protest this and say that it should ultimately be up to the end user to decide this.

Make the default how it is now, but give me an option to turn this off, I do NOT care if Various Arists are displayed as multiple albums.

I am currently digging through the ampache code trying to figure out how to do it.

**Comment by xekon** [comment]
The only easy fix I currently see is to adjust the Album for "Icon" and "Greatest Hits" to "Icon Scorpions", "Icon No Doubt", "Greatest Hits Doobie Brothers", "Greatest Hits Smashing Pumpkins"

This would Brute force fix the issue, allowing me to group my music the way that I want to, but it forces me to use incorrect album title.

**Comment by lusum** [comment]
A better solution, insert the album_artist tag with the name of the album artist, instead that only the artist tag, so the albums will be treated as different

**Comment by xekon** [comment]
I have both an album and an artist tag, are you saying that "album_artist" is another tag? is that part of the id3v2.3 spec?

I do see it is a configurable tag im the ampache webui. I will need to see if its a tag that eyeD3 is able to set. here is part of the bash script that I wrote that handles my tagging:

        printf "\rgenre:$genre artist:$artist year:$year album:$album disc:$disc track:$track title:$title\n"
        if [ -z "$disc" ]
        then
                #id3v2 --id3v2-only -a "$artist" -A "$album" -t "$title" -y "$year" -T "$track" "$f"
                echo eyeD3 --quiet --no-color -2 --to-v2.3 -a "$artist" -A "$album" -t "$title" -Y "$year" -n "$track" -G "$genreid" "$f"
        else
                #id3v2 --id3v2-only -a "$artist" -A "$album" -t "$title" -y "$year" -T "$track" --TPOS "$disc" "$f"
                echo eyeD3 --quiet --no-color -2 --to-v2.3 -a "$artist" -A "$album" -t "$title" -Y "$year" -n "$track" -d "$disc" -G "$genreid" "$f"
        fi

**Comment by wagnered** [comment]
The getID3 library  in Ampache reads TPE2 (band) for album artist for id3v2.3 frames and  ALBUMARTIST for vorbis comments.   I believe most gui taggers do also.

**Comment by xekon** [comment]
Thanks guys. My eyeD3 bash script is currently running adding the album_artist tag to all files, it may be hours before it done, will be sure to come back and close this if the album_artist tag resolves the problem (which I am sure it will). Thanks again :)

**Comment by xekon** [comment]
Every single mp3 has album_artist tag now, mediainfo shows both Performer and Album/Performer:

Complete name                            : /media/md0/audio/Alternative Rock/No Doubt/2010 - Icon/01 - Just A Girl.mp3
Album                                    : Icon
Album/Performer                          : No Doubt
Track name                               : Just A Girl
Track name/Position                      : 01
Performer                                : No Doubt
Genre                                    : Alternative Rock
Original/Released date                   : 2010
Recorded date                            : 2010

now that all mp3's have this tag I ran an update:

`sudo -u www-data php /var/www/ampache/bin/catalog_update.inc -c -v -a -g -m -o`

The problem I think now is because ampache has already scanned in these albums it showing as Artist: Various, despite the added tags. I was hoping the catalog update would see the changed tags and verify not just the mp3's but also the albums.

I am thinking I will have to wipe out the catalog and then recreate it, The main problem with this is I will lose user ratings, so I am going to create a backup of my system right now before I do anything.

![2020-12-30_16-46](https://user-images.githubusercontent.com/106644/103388608-cab47d80-4abe-11eb-91d0-c4203199a9e4.png)

**Comment by wagnered** [comment]
Ampache only stores the info in the db.  I can think only of TXXX, but it might mean a major change to the ID3v2.3 frame's write/display logic. I'm sure this feature would then be would probably be accessable only with Ampache..

**Comment by xekon** [comment]
I could manually go into each album in the ampache web ui then verify and manually change the the album_artist field.

**Comment by xekon** [comment]
I just got another idea, I will look for albums with this problem, remove only those from the directory, do a catalog update so that it removes them from the catalog, then re-add those albums and do another catalog update.

**Comment by xekon** [comment]
Simply making sure album, artist, and album_artist tags are in place and doing a catalog update is not enough.

The data in the ampache database will update for the individual songs, but the albums will still have the erroneous data in the db and list it as various Artist as well as both genres.

To fix the problem make sure all of your mp3s have the album, artist, and album_artist tags.

Next you need to remove the data from the databse for these albums. I did this by temporarily removing the albums.

In Ampache Web Interface browse All Albums.
go page by page, and search the page for "Various", make note of affected albums that are not supposed to be mixed together:

such as:
2010 - Icon - Scorpions
2010 - Icon - No Doubt
2001 - Greatest Hits - Smashing Pumpkins
2001 - Greatest Hits - The Doobie Brothers

temporarily move these albums to a folder outside of your catalog directories.

perform a scan which will remove the now missing albums from your ampache database.

`sudo -u www-data php /var/www/ampache/bin/catalog_update.inc -c`

Once this scan is complete, verify that the trouble albums are no longer listed in your database by browsing the albums in ampache web ui.

Now add a couple of the albums that were mixing together BACK to where you moved them from so that ampache can find them in catalog.

Now do another scan:

`sudo -u www-data php /var/www/ampache/bin/catalog_update.inc -c -v -a -g -o`

This should hopefully fix the issue, and the files should no longer be mixing as a single album, if so then you can put the remaining albums back in place, and perform a final scan.

Going forward make sure all tags are in place before adding to ampache and updating the catalog else face the turmoil associated with fixing the problem afterwards.

This was not a very user friendly issue to resolve, hopefully some more checks can be added to catalog_update for the album entries specifically so that after somebody updates/corrects the data in mp3 tags the data in the db can be updated easily.

**Comment by xekon** [comment]
Even though I fixed them from mixing, as shown in my previous screenshots, they do still appear on the same page, except now as separate disks... I would prefer even this to not happen, for instance I clicked into Icon for Scorpions, and you can see both No Doubt and Scorpions are on the page, but atleast the two are not mixed together.

If I click into "Scorpions" "Icon" album, is there a way ampache can show me JUST that album, instead of showing me all albums labled "Icon" for that year like it is currently doing:

![2020-12-30_18-48](https://user-images.githubusercontent.com/106644/103391810-09066880-4ad0-11eb-9dc9-34a4180c458e.png)

**Comment by xekon** [comment]
So now I understand, "album_artist" makes the situation somewhat better, but it definitely does not solve the problem. Ampache is still treating these two albums as the same album, all that the "album_artist" tag did was let ampache differentiate the two as separate disks. At least that appears to be what has happened.

**Comment by lusum** [comment]
1)add album_artist tag to every file of the album and save it

2)click update from tags

This should fix the problem

**Comment by xekon** [comment]
These albums were completely removed from the database and scanned back in with all files having the "album_artist" tag. However this problem is still here. Clicking update from tags simply says "No Update Needed". Here are pictures:

![2020-12-31_11-56](https://user-images.githubusercontent.com/106644/103424203-dc019680-4b5f-11eb-93bb-ff8b3c27b034.png)
![2020-12-31_11-57](https://user-images.githubusercontent.com/106644/103424202-db690000-4b5f-11eb-8138-fd2538d54df7.png)
![2020-12-31_11-58](https://user-images.githubusercontent.com/106644/103424201-dad06980-4b5f-11eb-8f11-33e8517ba68c.png)
![2020-12-31_11-59](https://user-images.githubusercontent.com/106644/103424200-da37d300-4b5f-11eb-9330-d72549c278a1.png)

**Comment by xekon** [comment]
@lusum additionally I restored my backup where the 2 albums were still mixed, and tried update from tags, it results in the same situation, the two albums get separated as different disks, but they still show up under the same album:
![2020-12-31_12-24](https://user-images.githubusercontent.com/106644/103424831-70b9c380-4b63-11eb-8909-c104b9636c65.png)
![2020-12-31_12-25](https://user-images.githubusercontent.com/106644/103424830-70212d00-4b63-11eb-9955-e710c60d65e3.png)
![2020-12-31_12-26](https://user-images.githubusercontent.com/106644/103424829-6f889680-4b63-11eb-845c-cc4d20a4c762.png)

**Comment by xekon** [comment]
The only thing I have not tried is wiping out the entire database and starting over.... but I really really really don't want to do that.

**Comment by lusum** [comment]
Ok, this is a bug, i never had a behaviour like this. The only thing i would suggest is to remove disk number, if only a disk is present.

**Comment by xekon** [comment]
@lusum none of these mp3's have a tagged disc number, ampache is just treating them this way anyway.

**Comment by lachlan-00** [comment]
I'm going back to work on Monday, is this "Album - Group multiple disks" on or off?

It seems to me like it's off just from browsing this but I'll look after the weekend.

When you update from tags the album check will test against columns like name, album_artist, etc etc. The check function can be very basic if a lot of tags are missing so I'm 99% sure that's the issue

It could be as simple as your tags aren't different enough for the album check function. When I get back I'll confirm what the check function looks for to determine if the album exists.

If check "finds" an album in the database it doesn't recreate it so I think it'll be easy enough to sort out.

**Comment by xekon** [comment]
In my settings "Album - Group multiple disks" is set to "On" which is what I want when disks are actually part of the same album.

however these two discs have both the "artist" tag and the "album_artist" and are still showing up under the same album, here is the current txt mediainfo for two of the files, the other files in the albums have the same tags except for track name and track position:

```
Complete name                            : /media/md0/audio/Rock-Alternative/No Doubt/2010 - Icon/01 - Just A Girl.mp3
Album                                    : Icon
Album/Performer                          : No Doubt
Track name                               : Just A Girl
Track name/Position                      : 01
Performer                                : No Doubt
Genre                                    : Alternative Rock
Original/Released date                   : 2010
Recorded date                            : 2010

Complete name                            : /media/md0/audio/Rock-Classic/Scorpions/2010 - Icon (HITS)/01 - Rock you like a hurricane.mp3
Album                                    : Icon
Album/Performer                          : Scorpions
Track name                               : Rock you like a hurricane
Track name/Position                      : 01
Performer                                : Scorpions
Genre                                    : Classic Rock
Original/Released date                   : 2010
Recorded date                            : 2010
```

**Comment by lachlan-00** [comment]
Cool, that's more than enough to go on tomorrow

**Comment by lachlan-00** [comment]
it could be that the albumartist isn't being found at the check time.

could you put this debug in the album class?
![image](https://user-images.githubusercontent.com/1305249/103503297-8a2e7880-4e9f-11eb-88cf-a21d2de2c249.png)


```
        debug_event('album.class', 'SQL: ' . $sql, 5);
        debug_event('album.class', 'PARAM: ' . $params, 5);
```

if you do it for both that will help
```
2021-01-04T15:12:58+10:00 [user] (album.class) -> SQL: SELECT MIN(`album`.`id`) AS `id` FROM `album` WHERE (`album`.`name` = ? OR LTRIM(CONCAT(COALESCE(`album`.`prefix`, ''), ' ', `album`.`name`)) = ?) AND `album`.`disk` = ? AND `album`.`year` = ? AND `album`.`mbid` = ? AND `album`.`album_artist` = ? AND `album`.`release_type` = ?
2021-01-04T15:12:58+10:00 [user] (album.class) -> PARAM: Array
2021-01-04T15:12:58+10:00 [user] (album.class) -> (
2021-01-04T15:12:58+10:00 [user] (album.class) ->     [0] => Snuff
2021-01-04T15:12:58+10:00 [user] (album.class) ->     [1] => Snuff
2021-01-04T15:12:58+10:00 [user] (album.class) ->     [2] => 1
2021-01-04T15:12:58+10:00 [user] (album.class) ->     [3] => 2002
2021-01-04T15:12:58+10:00 [user] (album.class) ->     [4] => 57aca4a0-1129-4a07-999b-756814f5d96f
2021-01-04T15:12:58+10:00 [user] (album.class) ->     [5] => 3006
2021-01-04T15:12:58+10:00 [user] (album.class) ->     [6] => ep
2021-01-04T15:12:58+10:00 [user] (album.class) -> )
2021-01-04T15:12:58+10:00 [user] (album.class) ->
```

**Comment by xekon** [comment]
Sure, I will work on this now.

**Comment by lachlan-00** [comment]
here's an odd one.
![image](https://user-images.githubusercontent.com/1305249/103505635-7a666280-4ea6-11eb-9fc2-f1f31e32d0e5.png)

why would that be a good idea!?

**Comment by lachlan-00** [comment]
can you comment out that line as well?

**Comment by xekon** [comment]
I found the section of code, adding the lines now and commenting out that other line, I need to have debugging enabled right?

**Comment by xekon** [comment]
not sure what you need so I just ran tail -f while doing it, here is the output:

```
2021-01-04T06:16:37+00:00 [jake] (ajax.server) -> Called for page: {browse} 
2021-01-04T06:16:37+00:00 [jake] (browse.ajax) -> Called for action: {get_filters} 
2021-01-04T06:16:41+00:00 [jake] (session.class) -> lvd01gjhi9ef1lvb5m11gbvtts has been extended to Mon, 04 Jan 2021 07:16:41 +0000 extension length 3600 
2021-01-04T06:16:41+00:00 [jake] (session.class) -> Session created: 393086071681b4af1bd0a4d1672d13e3 
2021-01-04T06:16:42+00:00 [jake] (browse.class) -> Show objects called for type {song} 
2021-01-04T06:16:42+00:00 [jake] (session.class) -> lvd01gjhi9ef1lvb5m11gbvtts has been extended to Mon, 04 Jan 2021 07:16:42 +0000 extension length 3600 
2021-01-04T06:16:43+00:00 [jake] (session.class) -> lvd01gjhi9ef1lvb5m11gbvtts has been extended to Mon, 04 Jan 2021 07:16:43 +0000 extension length 3600 
2021-01-04T06:16:43+00:00 [jake] (ajax.server) -> Called for page: {browse} 
2021-01-04T06:16:43+00:00 [jake] (browse.ajax) -> Called for action: {get_filters} 
2021-01-04T06:17:18+00:00 [ampache] (ajax.server) -> Called for page: {index} 
2021-01-04T06:17:25+00:00 [jake] (session.class) -> lvd01gjhi9ef1lvb5m11gbvtts has been extended to Mon, 04 Jan 2021 07:17:25 +0000 extension length 3600 
2021-01-04T06:17:25+00:00 [jake] (ajax.server) -> Called for page: {index} 
2021-01-04T06:17:34+00:00 [jake] (session.class) -> lvd01gjhi9ef1lvb5m11gbvtts has been extended to Mon, 04 Jan 2021 07:17:34 +0000 extension length 3600 
2021-01-04T06:17:34+00:00 [jake] (session.class) -> Session created: e5a390c7f256b1543d091a0a3ec32d19 
2021-01-04T06:17:34+00:00 [jake] (browse.class) -> Show objects called for type {album} 
2021-01-04T06:17:34+00:00 [jake] (session.class) -> lvd01gjhi9ef1lvb5m11gbvtts has been extended to Mon, 04 Jan 2021 07:17:34 +0000 extension length 3600 
2021-01-04T06:17:34+00:00 [jake] (ajax.server) -> Called for page: {index} 
2021-01-04T06:17:34+00:00 [jake] (session.class) -> lvd01gjhi9ef1lvb5m11gbvtts has been extended to Mon, 04 Jan 2021 07:17:34 +0000 extension length 3600 
2021-01-04T06:17:34+00:00 [jake] (ajax.server) -> Called for page: {browse} 
2021-01-04T06:17:34+00:00 [jake] (browse.ajax) -> Called for action: {get_filters} 
2021-01-04T06:17:36+00:00 [jake] (session.class) -> lvd01gjhi9ef1lvb5m11gbvtts has been extended to Mon, 04 Jan 2021 07:17:36 +0000 extension length 3600 
2021-01-04T06:17:36+00:00 [jake] (session.class) -> Session created: eac1ca5d65d36ee90963128329b83d50 
2021-01-04T06:17:36+00:00 [jake] (browse.class) -> Show objects called for type {song} 
2021-01-04T06:17:36+00:00 [jake] (session.class) -> lvd01gjhi9ef1lvb5m11gbvtts has been extended to Mon, 04 Jan 2021 07:17:36 +0000 extension length 3600 
2021-01-04T06:17:36+00:00 [jake] (ajax.server) -> Called for page: {browse} 
2021-01-04T06:17:36+00:00 [jake] (browse.ajax) -> Called for action: {get_filters} 
2021-01-04T06:17:41+00:00 [jake] (session.class) -> lvd01gjhi9ef1lvb5m11gbvtts has been extended to Mon, 04 Jan 2021 07:17:41 +0000 extension length 3600 
2021-01-04T06:17:41+00:00 [jake] (session.class) -> Session created: 7e178f23cd9160eb442224664a30c494 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Checking /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ _ Array on Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/01 - Just A Girl.mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ matched Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/01 - Just A Girl.mp3 on Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/01 - Just A Girl.mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Cleaning id3v2 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> Reading tags from /media/audio/Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/01 - Just A Girl.mp3 
2021-01-04T06:17:41+00:00 [jake] (album.class) -> SQL: SELECT MIN(`album`.`id`) AS `id` FROM `album` WHERE (`album`.`name` = ? OR LTRIM(CONCAT(COALESCE(`album`.`prefix`, ''), ' ', `album`.`name`)) = ?) AND `album`.`disk` = ? AND `album`.`year` = ? AND `album`.`mbid` IS NULL AND `album`.`album_artist` = ? AND `album`.`original_year` = ?  
2021-01-04T06:17:41+00:00 [jake] (album.class) -> PARAM: Array 
2021-01-04T06:17:41+00:00 [jake] (album.class) -> ( 
2021-01-04T06:17:41+00:00 [jake] (album.class) ->     [0] => Icon 
2021-01-04T06:17:41+00:00 [jake] (album.class) ->     [1] => Icon 
2021-01-04T06:17:41+00:00 [jake] (album.class) ->     [2] => 1 
2021-01-04T06:17:41+00:00 [jake] (album.class) ->     [3] => 2010 
2021-01-04T06:17:41+00:00 [jake] (album.class) ->     [4] => 246 
2021-01-04T06:17:41+00:00 [jake] (album.class) ->     [5] => 2010 
2021-01-04T06:17:41+00:00 [jake] (album.class) -> ) 
2021-01-04T06:17:41+00:00 [jake] (album.class) ->  
2021-01-04T06:17:41+00:00 [jake] (album.class) -> Album check created new album id 5883 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> migrate album from 4186 to 5883 
2021-01-04T06:17:41+00:00 [jake] (song.class) -> media-diff {"album":"OLD: 4186 --> 5883"} 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> /media/audio/Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/01 - Just A Girl.mp3 : differences found, updating database 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Checking /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ _ Array on Rock-Classic/Scorpions/2010 - Icon (HITS)/01 - Rock you like a hurricane.mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ matched Rock-Classic/Scorpions/2010 - Icon (HITS)/01 - Rock you like a hurricane.mp3 on Rock-Classic/Scorpions/2010 - Icon (HITS)/01 - Rock you like a hurricane.mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Cleaning id3v2 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> Reading tags from /media/audio/Rock-Classic/Scorpions/2010 - Icon (HITS)/01 - Rock you like a hurricane.mp3 
2021-01-04T06:17:41+00:00 [jake] (album.class) -> SQL: SELECT MIN(`album`.`id`) AS `id` FROM `album` WHERE (`album`.`name` = ? OR LTRIM(CONCAT(COALESCE(`album`.`prefix`, ''), ' ', `album`.`name`)) = ?) AND `album`.`disk` = ? AND `album`.`year` = ? AND `album`.`mbid` IS NULL AND `album`.`album_artist` = ? AND `album`.`original_year` = ?  
2021-01-04T06:17:41+00:00 [jake] (album.class) -> PARAM: Array 
2021-01-04T06:17:41+00:00 [jake] (album.class) -> ( 
2021-01-04T06:17:41+00:00 [jake] (album.class) ->     [0] => Icon 
2021-01-04T06:17:41+00:00 [jake] (album.class) ->     [1] => Icon 
2021-01-04T06:17:41+00:00 [jake] (album.class) ->     [2] => 1 
2021-01-04T06:17:41+00:00 [jake] (album.class) ->     [3] => 2010 
2021-01-04T06:17:41+00:00 [jake] (album.class) ->     [4] => 148 
2021-01-04T06:17:41+00:00 [jake] (album.class) ->     [5] => 2010 
2021-01-04T06:17:41+00:00 [jake] (album.class) -> ) 
2021-01-04T06:17:41+00:00 [jake] (album.class) ->  
2021-01-04T06:17:41+00:00 [jake] (album.class) -> Album check created new album id 5884 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> migrate album from 4186 to 5884 
2021-01-04T06:17:41+00:00 [jake] (art.class) -> duplicate... type:album old_id:4186 new_id:5884 
2021-01-04T06:17:41+00:00 [jake] (song.class) -> media-diff {"album":"OLD: 4186 --> 5884"} 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> /media/audio/Rock-Classic/Scorpions/2010 - Icon (HITS)/01 - Rock you like a hurricane.mp3 : differences found, updating database 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Checking /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ _ Array on Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/02 - It's My Life.mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ matched Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/02 - It's My Life.mp3 on Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/02 - It's My Life.mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Cleaning id3v2 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> Reading tags from /media/audio/Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/02 - It's My Life.mp3 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> migrate album from 4186 to 5883 
2021-01-04T06:17:41+00:00 [jake] (song.class) -> media-diff {"album":"OLD: 4186 --> 5883"} 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> /media/audio/Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/02 - It's My Life.mp3 : differences found, updating database 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Checking /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ _ Array on Rock-Classic/Scorpions/2010 - Icon (HITS)/02 - No one like you.mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ matched Rock-Classic/Scorpions/2010 - Icon (HITS)/02 - No one like you.mp3 on Rock-Classic/Scorpions/2010 - Icon (HITS)/02 - No one like you.mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Cleaning id3v2 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> Reading tags from /media/audio/Rock-Classic/Scorpions/2010 - Icon (HITS)/02 - No one like you.mp3 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> migrate album from 4186 to 5884 
2021-01-04T06:17:41+00:00 [jake] (art.class) -> duplicate... type:album old_id:4186 new_id:5884 
2021-01-04T06:17:41+00:00 [jake] (song.class) -> media-diff {"album":"OLD: 4186 --> 5884"} 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> /media/audio/Rock-Classic/Scorpions/2010 - Icon (HITS)/02 - No one like you.mp3 : differences found, updating database 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Checking /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ _ Array on Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/03 - Hey Baby (feat. Bounty Killer).mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ matched Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/03 - Hey Baby (feat. Bounty Killer).mp3 on Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/03 - Hey Baby (feat. Bounty Killer).mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Cleaning id3v2 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> Reading tags from /media/audio/Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/03 - Hey Baby (feat. Bounty Killer).mp3 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> migrate album from 4186 to 5883 
2021-01-04T06:17:41+00:00 [jake] (song.class) -> media-diff {"album":"OLD: 4186 --> 5883"} 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> /media/audio/Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/03 - Hey Baby (feat. Bounty Killer).mp3 : differences found, updating database 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Checking /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ _ Array on Rock-Classic/Scorpions/2010 - Icon (HITS)/03 - The Zoo.mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ matched Rock-Classic/Scorpions/2010 - Icon (HITS)/03 - The Zoo.mp3 on Rock-Classic/Scorpions/2010 - Icon (HITS)/03 - The Zoo.mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Cleaning id3v2 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> Reading tags from /media/audio/Rock-Classic/Scorpions/2010 - Icon (HITS)/03 - The Zoo.mp3 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> migrate album from 4186 to 5884 
2021-01-04T06:17:41+00:00 [jake] (art.class) -> duplicate... type:album old_id:4186 new_id:5884 
2021-01-04T06:17:41+00:00 [jake] (song.class) -> media-diff {"album":"OLD: 4186 --> 5884"} 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> /media/audio/Rock-Classic/Scorpions/2010 - Icon (HITS)/03 - The Zoo.mp3 : differences found, updating database 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Checking /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ _ Array on Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/04 - Bathwater.mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ matched Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/04 - Bathwater.mp3 on Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/04 - Bathwater.mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Cleaning id3v2 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> Reading tags from /media/audio/Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/04 - Bathwater.mp3 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> migrate album from 4186 to 5883 
2021-01-04T06:17:41+00:00 [jake] (song.class) -> media-diff {"album":"OLD: 4186 --> 5883"} 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> /media/audio/Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/04 - Bathwater.mp3 : differences found, updating database 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Checking /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ _ Array on Rock-Classic/Scorpions/2010 - Icon (HITS)/04 - Still Loving you.mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ matched Rock-Classic/Scorpions/2010 - Icon (HITS)/04 - Still Loving you.mp3 on Rock-Classic/Scorpions/2010 - Icon (HITS)/04 - Still Loving you.mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Cleaning id3v2 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> Reading tags from /media/audio/Rock-Classic/Scorpions/2010 - Icon (HITS)/04 - Still Loving you.mp3 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> migrate album from 4186 to 5884 
2021-01-04T06:17:41+00:00 [jake] (art.class) -> duplicate... type:album old_id:4186 new_id:5884 
2021-01-04T06:17:41+00:00 [jake] (song.class) -> media-diff {"album":"OLD: 4186 --> 5884"} 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> /media/audio/Rock-Classic/Scorpions/2010 - Icon (HITS)/04 - Still Loving you.mp3 : differences found, updating database 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Checking /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ _ Array on Rock-Classic/Scorpions/2010 - Icon (HITS)/05 - Big City Nights.mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ matched Rock-Classic/Scorpions/2010 - Icon (HITS)/05 - Big City Nights.mp3 on Rock-Classic/Scorpions/2010 - Icon (HITS)/05 - Big City Nights.mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Cleaning id3v2 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> Reading tags from /media/audio/Rock-Classic/Scorpions/2010 - Icon (HITS)/05 - Big City Nights.mp3 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> migrate album from 4186 to 5884 
2021-01-04T06:17:41+00:00 [jake] (art.class) -> duplicate... type:album old_id:4186 new_id:5884 
2021-01-04T06:17:41+00:00 [jake] (song.class) -> media-diff {"album":"OLD: 4186 --> 5884"} 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> /media/audio/Rock-Classic/Scorpions/2010 - Icon (HITS)/05 - Big City Nights.mp3 : differences found, updating database 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Checking /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ _ Array on Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/05 - Sunday Morning.mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ matched Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/05 - Sunday Morning.mp3 on Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/05 - Sunday Morning.mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Cleaning id3v2 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> Reading tags from /media/audio/Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/05 - Sunday Morning.mp3 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> migrate album from 4186 to 5883 
2021-01-04T06:17:41+00:00 [jake] (song.class) -> media-diff {"album":"OLD: 4186 --> 5883"} 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> /media/audio/Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/05 - Sunday Morning.mp3 : differences found, updating database 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Checking /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ _ Array on Rock-Classic/Scorpions/2010 - Icon (HITS)/06 - Believe in Love.mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ matched Rock-Classic/Scorpions/2010 - Icon (HITS)/06 - Believe in Love.mp3 on Rock-Classic/Scorpions/2010 - Icon (HITS)/06 - Believe in Love.mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Cleaning id3v2 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> Reading tags from /media/audio/Rock-Classic/Scorpions/2010 - Icon (HITS)/06 - Believe in Love.mp3 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> migrate album from 4186 to 5884 
2021-01-04T06:17:41+00:00 [jake] (art.class) -> duplicate... type:album old_id:4186 new_id:5884 
2021-01-04T06:17:41+00:00 [jake] (song.class) -> media-diff {"album":"OLD: 4186 --> 5884"} 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> /media/audio/Rock-Classic/Scorpions/2010 - Icon (HITS)/06 - Believe in Love.mp3 : differences found, updating database 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Checking /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ _ Array on Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/06 - Hella Good.mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ matched Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/06 - Hella Good.mp3 on Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/06 - Hella Good.mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Cleaning id3v2 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> Reading tags from /media/audio/Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/06 - Hella Good.mp3 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> migrate album from 4186 to 5883 
2021-01-04T06:17:41+00:00 [jake] (song.class) -> media-diff {"album":"OLD: 4186 --> 5883"} 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> /media/audio/Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/06 - Hella Good.mp3 : differences found, updating database 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Checking /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ _ Array on Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/07 - New.mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ matched Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/07 - New.mp3 on Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/07 - New.mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Cleaning id3v2 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> Reading tags from /media/audio/Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/07 - New.mp3 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> migrate album from 4186 to 5883 
2021-01-04T06:17:41+00:00 [jake] (song.class) -> media-diff {"album":"OLD: 4186 --> 5883"} 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> /media/audio/Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/07 - New.mp3 : differences found, updating database 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Checking /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ _ Array on Rock-Classic/Scorpions/2010 - Icon (HITS)/07 - Rhythm of love.mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ matched Rock-Classic/Scorpions/2010 - Icon (HITS)/07 - Rhythm of love.mp3 on Rock-Classic/Scorpions/2010 - Icon (HITS)/07 - Rhythm of love.mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Cleaning id3v2 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> Reading tags from /media/audio/Rock-Classic/Scorpions/2010 - Icon (HITS)/07 - Rhythm of love.mp3 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> migrate album from 4186 to 5884 
2021-01-04T06:17:41+00:00 [jake] (art.class) -> duplicate... type:album old_id:4186 new_id:5884 
2021-01-04T06:17:41+00:00 [jake] (song.class) -> media-diff {"album":"OLD: 4186 --> 5884"} 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> /media/audio/Rock-Classic/Scorpions/2010 - Icon (HITS)/07 - Rhythm of love.mp3 : differences found, updating database 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Checking /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ _ Array on Rock-Classic/Scorpions/2010 - Icon (HITS)/08 - I can't explain.mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ matched Rock-Classic/Scorpions/2010 - Icon (HITS)/08 - I can't explain.mp3 on Rock-Classic/Scorpions/2010 - Icon (HITS)/08 - I can't explain.mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Cleaning id3v2 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> Reading tags from /media/audio/Rock-Classic/Scorpions/2010 - Icon (HITS)/08 - I can't explain.mp3 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> migrate album from 4186 to 5884 
2021-01-04T06:17:41+00:00 [jake] (art.class) -> duplicate... type:album old_id:4186 new_id:5884 
2021-01-04T06:17:41+00:00 [jake] (song.class) -> media-diff {"album":"OLD: 4186 --> 5884"} 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> /media/audio/Rock-Classic/Scorpions/2010 - Icon (HITS)/08 - I can't explain.mp3 : differences found, updating database 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Checking /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ _ Array on Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/08 - Underneath It All (feat. Lady Saw].mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ matched Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/08 - Underneath It All (feat. Lady Saw].mp3 on Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/08 - Underneath It All (feat. Lady Saw].mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Cleaning id3v2 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> Reading tags from /media/audio/Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/08 - Underneath It All (feat. Lady Saw].mp3 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> migrate album from 4186 to 5883 
2021-01-04T06:17:41+00:00 [jake] (song.class) -> media-diff {"album":"OLD: 4186 --> 5883"} 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> /media/audio/Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/08 - Underneath It All (feat. Lady Saw].mp3 : differences found, updating database 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Checking /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ _ Array on Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/09 - Excuse Me Mr..mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ matched Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/09 - Excuse Me Mr..mp3 on Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/09 - Excuse Me Mr..mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Cleaning id3v2 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> Reading tags from /media/audio/Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/09 - Excuse Me Mr..mp3 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> migrate album from 4186 to 5883 
2021-01-04T06:17:41+00:00 [jake] (song.class) -> media-diff {"album":"OLD: 4186 --> 5883"} 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> /media/audio/Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/09 - Excuse Me Mr..mp3 : differences found, updating database 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Checking /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ _ Array on Rock-Classic/Scorpions/2010 - Icon (HITS)/09 - Tease me, Please Me.mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ matched Rock-Classic/Scorpions/2010 - Icon (HITS)/09 - Tease me, Please Me.mp3 on Rock-Classic/Scorpions/2010 - Icon (HITS)/09 - Tease me, Please Me.mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Cleaning id3v2 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> Reading tags from /media/audio/Rock-Classic/Scorpions/2010 - Icon (HITS)/09 - Tease me, Please Me.mp3 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> migrate album from 4186 to 5884 
2021-01-04T06:17:41+00:00 [jake] (art.class) -> duplicate... type:album old_id:4186 new_id:5884 
2021-01-04T06:17:41+00:00 [jake] (song.class) -> media-diff {"album":"OLD: 4186 --> 5884"} 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> /media/audio/Rock-Classic/Scorpions/2010 - Icon (HITS)/09 - Tease me, Please Me.mp3 : differences found, updating database 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Checking /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ _ Array on Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/10 - Running.mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ matched Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/10 - Running.mp3 on Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/10 - Running.mp3 
2021-01-04T06:17:41+00:00 [jake] (vainfo.class) -> Cleaning id3v2 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> Reading tags from /media/audio/Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/10 - Running.mp3 
2021-01-04T06:17:41+00:00 [jake] (catalog.class) -> migrate album from 4186 to 5883 
2021-01-04T06:17:42+00:00 [jake] (song.class) -> media-diff {"album":"OLD: 4186 --> 5883"} 
2021-01-04T06:17:42+00:00 [jake] (catalog.class) -> /media/audio/Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/10 - Running.mp3 : differences found, updating database 
2021-01-04T06:17:42+00:00 [jake] (vainfo.class) -> Checking /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ _ Array on Rock-Classic/Scorpions/2010 - Icon (HITS)/10 - Wind of Change.mp3 
2021-01-04T06:17:42+00:00 [jake] (vainfo.class) -> /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ matched Rock-Classic/Scorpions/2010 - Icon (HITS)/10 - Wind of Change.mp3 on Rock-Classic/Scorpions/2010 - Icon (HITS)/10 - Wind of Change.mp3 
2021-01-04T06:17:42+00:00 [jake] (vainfo.class) -> Cleaning id3v2 
2021-01-04T06:17:42+00:00 [jake] (catalog.class) -> Reading tags from /media/audio/Rock-Classic/Scorpions/2010 - Icon (HITS)/10 - Wind of Change.mp3 
2021-01-04T06:17:42+00:00 [jake] (catalog.class) -> migrate album from 4186 to 5884 
2021-01-04T06:17:42+00:00 [jake] (art.class) -> duplicate... type:album old_id:4186 new_id:5884 
2021-01-04T06:17:42+00:00 [jake] (song.class) -> media-diff {"album":"OLD: 4186 --> 5884"} 
2021-01-04T06:17:42+00:00 [jake] (catalog.class) -> /media/audio/Rock-Classic/Scorpions/2010 - Icon (HITS)/10 - Wind of Change.mp3 : differences found, updating database 
2021-01-04T06:17:42+00:00 [jake] (vainfo.class) -> Checking /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ _ Array on Rock-Classic/Scorpions/2010 - Icon (HITS)/11 - Humanity.mp3 
2021-01-04T06:17:42+00:00 [jake] (vainfo.class) -> /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ matched Rock-Classic/Scorpions/2010 - Icon (HITS)/11 - Humanity.mp3 on Rock-Classic/Scorpions/2010 - Icon (HITS)/11 - Humanity.mp3 
2021-01-04T06:17:42+00:00 [jake] (vainfo.class) -> Cleaning id3v2 
2021-01-04T06:17:42+00:00 [jake] (catalog.class) -> Reading tags from /media/audio/Rock-Classic/Scorpions/2010 - Icon (HITS)/11 - Humanity.mp3 
2021-01-04T06:17:42+00:00 [jake] (catalog.class) -> migrate album from 4186 to 5884 
2021-01-04T06:17:42+00:00 [jake] (art.class) -> duplicate... type:album old_id:4186 new_id:5884 
2021-01-04T06:17:42+00:00 [jake] (song.class) -> media-diff {"album":"OLD: 4186 --> 5884"} 
2021-01-04T06:17:42+00:00 [jake] (catalog.class) -> /media/audio/Rock-Classic/Scorpions/2010 - Icon (HITS)/11 - Humanity.mp3 : differences found, updating database 
2021-01-04T06:17:42+00:00 [jake] (vainfo.class) -> Checking /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ _ Array on Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/11 - Spiderwebs.mp3 
2021-01-04T06:17:42+00:00 [jake] (vainfo.class) -> /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ matched Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/11 - Spiderwebs.mp3 on Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/11 - Spiderwebs.mp3 
2021-01-04T06:17:42+00:00 [jake] (vainfo.class) -> Cleaning id3v2 
2021-01-04T06:17:42+00:00 [jake] (catalog.class) -> Reading tags from /media/audio/Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/11 - Spiderwebs.mp3 
2021-01-04T06:17:42+00:00 [jake] (catalog.class) -> migrate album from 4186 to 5883 
2021-01-04T06:17:42+00:00 [jake] (song.class) -> media-diff {"album":"OLD: 4186 --> 5883"} 
2021-01-04T06:17:42+00:00 [jake] (catalog.class) -> /media/audio/Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/11 - Spiderwebs.mp3 : differences found, updating database 
2021-01-04T06:17:42+00:00 [jake] (vainfo.class) -> Checking /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ _ Array on Rock-Classic/Scorpions/2010 - Icon (HITS)/12 - Raised On Rock.mp3 
2021-01-04T06:17:42+00:00 [jake] (vainfo.class) -> /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ matched Rock-Classic/Scorpions/2010 - Icon (HITS)/12 - Raised On Rock.mp3 on Rock-Classic/Scorpions/2010 - Icon (HITS)/12 - Raised On Rock.mp3 
2021-01-04T06:17:42+00:00 [jake] (vainfo.class) -> Cleaning id3v2 
2021-01-04T06:17:42+00:00 [jake] (catalog.class) -> Reading tags from /media/audio/Rock-Classic/Scorpions/2010 - Icon (HITS)/12 - Raised On Rock.mp3 
2021-01-04T06:17:42+00:00 [jake] (catalog.class) -> migrate album from 4186 to 5884 
2021-01-04T06:17:42+00:00 [jake] (art.class) -> duplicate... type:album old_id:4186 new_id:5884 
2021-01-04T06:17:42+00:00 [jake] (song.class) -> media-diff {"album":"OLD: 4186 --> 5884"} 
2021-01-04T06:17:42+00:00 [jake] (catalog.class) -> /media/audio/Rock-Classic/Scorpions/2010 - Icon (HITS)/12 - Raised On Rock.mp3 : differences found, updating database 
2021-01-04T06:17:42+00:00 [jake] (vainfo.class) -> Checking /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ _ Array on Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/12 - Simple Kind Of Life.mp3 
2021-01-04T06:17:42+00:00 [jake] (vainfo.class) -> /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ matched Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/12 - Simple Kind Of Life.mp3 on Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/12 - Simple Kind Of Life.mp3 
2021-01-04T06:17:42+00:00 [jake] (vainfo.class) -> Cleaning id3v2 
2021-01-04T06:17:42+00:00 [jake] (catalog.class) -> Reading tags from /media/audio/Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/12 - Simple Kind Of Life.mp3 
2021-01-04T06:17:42+00:00 [jake] (catalog.class) -> migrate album from 4186 to 5883 
2021-01-04T06:17:42+00:00 [jake] (song.class) -> media-diff {"album":"OLD: 4186 --> 5883"} 
2021-01-04T06:17:42+00:00 [jake] (catalog.class) -> /media/audio/Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/12 - Simple Kind Of Life.mp3 : differences found, updating database 
2021-01-04T06:17:42+00:00 [jake] (vainfo.class) -> Checking /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ _ Array on Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/13 - Don't Speak.mp3 
2021-01-04T06:17:42+00:00 [jake] (vainfo.class) -> /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ matched Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/13 - Don't Speak.mp3 on Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/13 - Don't Speak.mp3 
2021-01-04T06:17:42+00:00 [jake] (vainfo.class) -> Cleaning id3v2 
2021-01-04T06:17:42+00:00 [jake] (catalog.class) -> Reading tags from /media/audio/Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/13 - Don't Speak.mp3 
2021-01-04T06:17:42+00:00 [jake] (catalog.class) -> migrate album from 4186 to 5883 
2021-01-04T06:17:42+00:00 [jake] (song.class) -> media-diff {"album":"OLD: 4186 --> 5883"} 
2021-01-04T06:17:42+00:00 [jake] (catalog.class) -> /media/audio/Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/13 - Don't Speak.mp3 : differences found, updating database 
2021-01-04T06:17:42+00:00 [jake] (vainfo.class) -> Checking /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ _ Array on Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/14 - Ex Girlfriend.mp3 
2021-01-04T06:17:42+00:00 [jake] (vainfo.class) -> /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ matched Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/14 - Ex Girlfriend.mp3 on Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/14 - Ex Girlfriend.mp3 
2021-01-04T06:17:42+00:00 [jake] (vainfo.class) -> Cleaning id3v2 
2021-01-04T06:17:42+00:00 [jake] (catalog.class) -> Reading tags from /media/audio/Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/14 - Ex Girlfriend.mp3 
2021-01-04T06:17:42+00:00 [jake] (catalog.class) -> migrate album from 4186 to 5883 
2021-01-04T06:17:42+00:00 [jake] (song.class) -> media-diff {"album":"OLD: 4186 --> 5883"} 
2021-01-04T06:17:42+00:00 [jake] (catalog.class) -> /media/audio/Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/14 - Ex Girlfriend.mp3 : differences found, updating database 
2021-01-04T06:17:42+00:00 [jake] (vainfo.class) -> Checking /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ _ Array on Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/15 - Trapped In A Box.mp3 
2021-01-04T06:17:42+00:00 [jake] (vainfo.class) -> /(.+?)\/(.+?)\/([0-9]+?)\s\-\s(.+?)\/([0-9]+?)\s\-\s(.+?)\..+$/ matched Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/15 - Trapped In A Box.mp3 on Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/15 - Trapped In A Box.mp3 
2021-01-04T06:17:42+00:00 [jake] (vainfo.class) -> Cleaning id3v2 
2021-01-04T06:17:42+00:00 [jake] (catalog.class) -> Reading tags from /media/audio/Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/15 - Trapped In A Box.mp3 
2021-01-04T06:17:42+00:00 [jake] (catalog.class) -> migrate album from 4186 to 5883 
2021-01-04T06:17:42+00:00 [jake] (song.class) -> media-diff {"album":"OLD: 4186 --> 5883"} 
2021-01-04T06:17:42+00:00 [jake] (catalog.class) -> /media/audio/Rock-Alternative/No Doubt/2010 - Icon (Compilation Album)/15 - Trapped In A Box.mp3 : differences found, updating database 
2021-01-04T06:17:45+00:00 [jake] (session.class) -> lvd01gjhi9ef1lvb5m11gbvtts has been extended to Mon, 04 Jan 2021 07:17:45 +0000 extension length 3600 
2021-01-04T06:17:45+00:00 [jake] (session.class) -> Session created: 9b3f9f8a81724d4ed232afdb443c7581 
2021-01-04T06:17:45+00:00 [jake] (browse.class) -> Show objects called for type {song} 
2021-01-04T06:17:45+00:00 [jake] (browse.class) -> Show objects called for type {song} 
2021-01-04T06:18:18+00:00 [ampache] (ajax.server) -> Called for page: {index} 
2021-01-04T06:18:25+00:00 [jake] (session.class) -> lvd01gjhi9ef1lvb5m11gbvtts has been extended to Mon, 04 Jan 2021 07:18:25 +0000 extension length 3600 
2021-01-04T06:18:25+00:00 [jake] (ajax.server) -> Called for page: {index}
```

**Comment by lachlan-00** [comment]
okay, so i see that there are two objects now, there's an album 5883 and 5884

Both migrated from 4186 into separate items! do you see separate albums in each artist now?

**Comment by xekon** [comment]
If I browse the albums for both scorpions and nodoubt, they both have the album Icon listed, however clicking into the album lists both discs, regardless if I click in from no doubt or scorpions:

![2021-01-03_22-29](https://user-images.githubusercontent.com/106644/103507210-7b01e900-4e13-11eb-958d-c07d332dca90.png)

![2021-01-03_22-30](https://user-images.githubusercontent.com/106644/103507217-7fc69d00-4e13-11eb-8aca-5b41650c7512.png)

![2021-01-03_22-30_1](https://user-images.githubusercontent.com/106644/103507225-848b5100-4e13-11eb-84b3-6e7138cc52c3.png)

**Comment by xekon** [comment]
If I click in to "Icon" from scorpions then the address bar shows:

https://amp.solosaga.com/index.php#albums.php?action=show&album=5884

If I click in to "Icon" from no doubt then the address bar shows:

https://amp.solosaga.com/index.php#albums.php?action=show&album=5883

**Comment by xekon** [comment]
If I click in from No doubt then address bar shows:

https://amp.example.com/index.php#albums.php?action=show&album=5883

If I click in from Scorpions then address bar shows:

https://amp.example.com/index.php#albums.php?action=show&album=5884

however in both albums, both discs are listed.

**Comment by xekon** [comment]
If you need I can send you the first track of each disc, if it would help with debugging.

Or if you need me to enter any more debug lines I can do that as well.

**Comment by lachlan-00** [comment]
Now that we've fixed the check error there's a sql query done to pull in each disk of an album on full grouping so it means there's an issue there.

We're half way there now! can you add another sql debug in here (album.class again)?
![image](https://user-images.githubusercontent.com/1305249/103508015-19418d80-4eac-11eb-8fe5-a82b44d52729.png)

just the $sql now
```
debug_event('album.class', 'SQL: ' . $sql, 5);
```

**Comment by xekon** [comment]
one moment, doing it now.

**Comment by lachlan-00** [comment]
it doesn't do any check for album artist which isn't what i'd want it to do.

**Comment by lachlan-00** [comment]
i'll be back on later tomorrow but i think that's the last piece of the puzzle!

**Comment by xekon** [comment]
2021-01-04T06:58:42+00:00 [jake] (album.class) -> SQL: SELECT DISTINCT `album`.`id`, MAX(`album`.`disk`) AS `disk` FROM `album` LEFT JOIN `song` ON `song`.`album`=`album`.`id` LEFT JOIN `catalog` ON `catalog`.`id` = `song`.`catalog` WHERE `album`.`mbid` is null AND `album`.`release_type` is null AND (`album`.`name` = 'Icon' OR LTRIM(CONCAT(COALESCE(`album`.`prefix`, ''), ' ', `album`.`name`)) = 'Icon') AND `album`.`year` = 2010   GROUP BY `album`.`id` ORDER BY `album`.`disk` ASC

**Comment by lachlan-00** [comment]
Awesome I'm on family time now but I'll be able to sort this first thing in the morning.

Thanks for the patience while I was on break!

**Comment by xekon** [comment]
You rock, thanks @lachlan-00 I am just super relieved your looking at it :)

**Comment by lachlan-00** [comment]
f156ef2cd should cover you now.

If your album didn't have an mbid it was **very** loosely joining other disks by name/year.

joining by album artist is something we should have done earlier but i hadn't noticed as 80% of my stuff is tagged in picard.

**Comment by xekon** [comment]
YES! You fixed it! Thank you so much! Working perfect now.

**Comment by xekon** [comment]
@lachlan-00 is there a timeframe before this gets merged into stable? I know things like this are usually it happens whenever it happens, but figured don't hurt to ask for a general idea, such as within 6 months or probably not for atleast another year.

I ask because I am considering running stable instead of develop so that I can make use of the exciting ampache kodi plugin I just found: https://github.com/ampache/ampache-xbmc-plugin

I opened a few bug reports but learned that the reason I am having issues is because of incompatibility with the develop branch of ampache:

https://github.com/ampache/ampache-xbmc-plugin/issues/7
https://github.com/ampache/ampache-xbmc-plugin/issues/8
https://github.com/ampache/ampache-xbmc-plugin/issues/9
https://github.com/ampache/ampache-xbmc-plugin/issues/10

I have been a regular user of Kodi going all the way back to when it was called XBMC, I first used it on my original XBOX! So to find out there was a plugin to use ampache server as a media source in Kodi was quite exciting, it makes listening to my music on my TV's sound system very easy!

There have been some fixes you have done in develop that I really don't want to live without if I switched to the stable branch:
https://github.com/ampache/ampache/issues/2667
https://github.com/ampache/ampache/issues/2310
https://github.com/ampache/ampache/issues/2304

**Comment by lachlan-00** [comment]
I'm in the process of 4.3.0 so I'll check those out to make sure they're in.

**Comment by xekon** [comment]
RIGHT ON!

**Comment by lachlan-00** [comment]
To fix playlists in Kodi I'm going to put an option in the database for hiding searches from the API.

But that will be a 4.4.0 option

**Comment by xekon** [comment]
Will that make it so that the smart playlists will work from the kodi plugin?

**Comment by lachlan-00** [comment]
> @lachlan-00 Will that make it so that the smart playlists will work from the kodi plugin?

It will fix regular playlists but not searches for the moment as a stopgap

**Comment by xekon** [comment]
ah I see, so I assume that means that smart playlists use searches. Maybe I can find a way to duplicate a smart playlist into a regular playlist periodically, so that I can play those too, I appreciate the explanation. Thanks for working on this.

**Comment by lachlan-00** [comment]
> ah I see, so I assume that means that smart playlists use searches. Maybe I can find a way to duplicate a smart playlist into a regular playlist periodically, so that I can play those too, I appreciate the explanation. Thanks for working on this.

That would actually make a good CLI command. Caching searches into a smartlist would actually be a pretty good idea.

Then run the command maybe in Cron to recreate.

**Comment by lusum** [comment]
Genre is not a problem, i will add in the future, when i will have a 5.0 server to test the code, about smart playlist i have to rewrite a big part of the code, i think i will in the next future, but for now, i prefet to make the kodi official version as much stable as possible. So i prefer to skip, for now, smart playlist and not add untested code.

**Comment by lachlan-00** [comment]
> There have been some fixes you have done in develop that I really don't want to live without if I switched to the stable branch:

just confirming that these are already part of master
#2667 - queries are fixed
#2310 - the averages e.g. "(average 4)" is shown separately from the ratings
#2304 - artist art from last.fm is ignored and spotify search is in

For a kodi user stick with stable. @lusum has put so much effort into this plugin and the ampache4 branch has had 2 years of work (so far) on it so it's really the best option.

I'm going to work on a config/database option to separate playlists and smartlists for this as well to make sure that the current kodi plugin can list playlists without a code change.

before we make develop go crazy there will be updates to allow keeping the old stable/develop anyway so i'll try and keep as many people aware of the incoming change before we do it.

**Comment by xekon** [comment]
@lachlan-00 Thanks for checking on those for me, I am going to try and move from Ampache develop to Ampache Stable, but not sure if I will have any trouble because of the newer version of my Ampache Database.

I will make a backup first and then give it a try.

**Comment by lusum** [comment]
> @lachlan-00 Thanks for checking on those for me, I am going to try and move from Ampache develop to Ampache Stable, but not sure if I will have any trouble because of the newer version of my Ampache Database.
> 
> I will make a backup first and then give it a try.

You can install two different subfolders on your http server. Like 127.0.0.1/ampache and 127.0.0.1/ampache-dev, i have to do this work, but i have little time to setup the server. Maintain two different kodi version is another beast, i need a virtual machine to test both ( 18 and 19 )

**Comment by lachlan-00** [comment]
The biggest diff between master/Dev will be the custom_datetime changes.

If you use the same database that's the only thing you should need to change

Option for playlists incoming. I might also check out doing this as a client option.
![20210205_084022.jpg](https://user-images.githubusercontent.com/1305249/106964581-0b727700-678e-11eb-8508-a091e7cca654.jpg)

Do you send version in the client string @lusum ? I'll check it out myself from this server

**Comment by lusum** [comment]
>Do you send version in the client string @lusum ? I'll check it out myself from this server

No, i check the server api version and the client adapt his behaviour based on server version.

**Comment by lachlan-00** [comment]
That's okay i've put in some simple workarounds for now in 4.3.0 i've set 2 ways to hide the smart lists.

In the playlists / get_indexes calls set hide_search=1
```
|'hide_search'|integer   | 0,1, if true do not include searches/smartlists in the result |YES     |
```

in the config for develop and 4.3.0
```
; Hide Searches from the API
; Set this to true and the get_indexes and playlists methods will not automatically include
; searches/smart lists in the results.
; Older versions of the Kodi Add-on will require this for playlists to work
; DEFAULT: "false"
;hide_search = "true"
```

**Comment by lusum** [comment]
Good! I have already made an hide searches in my kodi plugin, but it will be present in the next version
